### PR TITLE
todo using use context

### PR DESCRIPTION
**Pull request by abhi1906singh**

### CODE REVIEW

File: src/comonents/InputForm.jsx [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
-      
+    <div style={{ width: "100%", display: "flex" }}>
+      <input
+        style={{ width: "100%" }}
+        type="text"
+        placeholder="Enter your text"
+        value={todo}
+        onChange={handleChange}
+      />
+      <ButtonPage />
```
Comment (Kailash121212) [inline src/comonents/InputForm.jsx]: use appropriate name for component, this is not page,
use like button.

File: src/comonents/InputForm.jsx [hunk 2]
```diff
@@ -1,11 +1,21 @@
-import React from 'react'
+import React, { useContext } from "react";
+import ButtonPage from "./ButtonPage";
+import { MyNoteContext } from "../context/NoteState";
 
 function InputForm() {
```
Comment (Kailash121212) [inline src/comonents/InputForm.jsx]: this component name should be InputField, not form,
this represent particular input field.

File: src/comonents/ButtonPage.jsx [hunk 3]
```diff
@@ -1,9 +1,11 @@
-import React from 'react'
+import React, { useContext } from 'react'
+import { MyNoteContext } from '../context/NoteState'
 
 function ButtonPage() {
```
Comment (Kailash121212) [inline src/comonents/ButtonPage.jsx]: this component name should be  Button, not page,
this represent particular button component.

File: src/comonents/ButtonPage.jsx [hunk 4]
```diff
@@ -1,9 +1,11 @@
...
+  const {handleClick}=useContext(MyNoteContext);
   return (
     <div>
-      
+      <button style={{background:"#2222c9",color:"#ffffff"}} onClick={handleClick}>Submit</button>
```
Comment (Kailash121212) [inline src/comonents/ButtonPage.jsx]: remove inline css, create seperate file for css and attach

File: src/comonents/InputForm.jsx [hunk 5]
```diff
@@ -1,11 +1,21 @@
...
+  const { todo, handleChange } = useContext(MyNoteContext);
   return (
-    <div>
-      
+    <div style={{ width: "100%", display: "flex" }}>
```
Comment (Kailash121212) [inline src/comonents/InputForm.jsx]: remove inline css, create separate file for css and attach

File: src/pages/HomePage.jsx [hunk 6]
```diff
@@ -1,11 +1,16 @@
-import React from 'react'
+import React from "react";
+import NoteState from "../context/NoteState";
+import InputForm from "../comonents/InputForm";
 
 function HomePage() {
   return (
-    <div>
-      
+    <div style={{width:"60%",margin:'auto'}}>
```
Comment (Kailash121212) [inline src/pages/HomePage.jsx]: remove inline css, create seperate file for css and attach

File: src/pages/TablePage.jsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import React, { useContext } from 'react'
+import { MyNoteContext } from '../context/NoteState'
+
+function TablePage() {
+    var {arr}=useContext(MyNoteContext);
+    console.log(arr);
```
Comment (Kailash121212) [inline src/pages/TablePage.jsx]: remove console

File: src/pages/HomePage.jsx [hunk 8]
```diff
@@ -1,11 +1,16 @@
-import React from 'react'
+import React from "react";
+import NoteState from "../context/NoteState";
+import InputForm from "../comonents/InputForm";
 
 function HomePage() {
```
Comment (Kailash121212) [inline src/pages/HomePage.jsx]: this is main page,
you can use name dashboard

File: src/pages/TablePage.jsx [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import React, { useContext } from 'react'
+import { MyNoteContext } from '../context/NoteState'
+
+function TablePage() {
```
Comment (Kailash121212) [inline src/pages/TablePage.jsx]: tablepage do not represent separate page , this is part of main page which is home page.
so make it as  a component of home page.

File: src/comonents/InputForm.jsx (hunk 2 above)
Comment (abhi1906singh) [reply src/comonents/InputForm.jsx]: Changed

File: src/comonents/ButtonPage.jsx (hunk 3 above)
Comment (abhi1906singh) [reply src/comonents/ButtonPage.jsx]: Changed sir

File: src/pages/TablePage.jsx (hunk 7 above)
Comment (abhi1906singh) [reply src/pages/TablePage.jsx]: Removed

File: src/pages/TablePage.jsx (hunk 9 above)
Comment (abhi1906singh) [reply src/pages/TablePage.jsx]: Resolved sir

File: src/comonents/ButtonPage.jsx (hunk 4 above)
Comment (abhi1906singh) [reply src/comonents/ButtonPage.jsx]: Changed sir

File: src/pages/HomePage.jsx (hunk 6 above)
Comment (abhi1906singh) [reply src/pages/HomePage.jsx]: Resolved sir

File: src/comonents/InputForm.jsx (hunk 5 above)
Comment (abhi1906singh) [reply src/comonents/InputForm.jsx]: Resolved sir

File: src/comonents/InputForm.jsx (hunk 1 above)
Comment (abhi1906singh) [reply src/comonents/InputForm.jsx]: Resolved sir

File: src/comonents/InputField.jsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  return (
+    <div className="input_div">
+      <input
+        className="text_todo"
+        type="text"
+        placeholder="Enter your text"
+        value={inputText}
+        onChange={handleChange}
+      />
+      <Button />
```
Comment (Kailash121212) [inline src/comonents/InputField.jsx]: remove button from here
this component represent input field

File: src/comonents/TablePage.jsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
+import React, { useContext } from "react";
+import { MyNoteContext } from "../context/NoteState";
+import "./TablePage.css";
+
+function TablePage() {
+  var { todosList,delete_Click,edit_Click } = useContext(MyNoteContext);
+  console.log(todosList,"hbbbbbbbbb");
```
Comment (Kailash121212) [inline src/comonents/TablePage.jsx]: remove console

File: src/context/NoteState.jsx [hunk 12]
```diff
@@ -1,11 +1,52 @@
...
+    setInputText("");
+  };
+
+
+  const delete_Click = (id) => {
+    const updatedItems = todosList.filter(item => item.id !== id);
+    setTodosList(updatedItems);
+  };
+
+  // const edit_Click=(id)=>{
```
Comment (Kailash121212) [inline src/context/NoteState.jsx]: remove commented code

File: src/pages/HomeDashBoard.jsx [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+import NoteState from "../context/NoteState";
+import Inputfield from "../comonents/InputField";
+import TablePage from "../comonents/TablePage";
+import './homepage.css'
+
+function HomeDashBoard() {
+  return (
+    <div>
+      <div className="home_div">
+        <NoteState>
```
Comment (Kailash121212) [inline src/pages/HomeDashBoard.jsx]: context wrapper should create in app.js file
if why have more the one page.
we will not able to access this.

File: src/pages/HomeDashBoard.jsx [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+          <h1>ToDo App</h1>
+          <Inputfield />
```
Comment (Kailash121212) [inline src/pages/HomeDashBoard.jsx]: input field represent particular input field , 
make here a form and use onSubmit property.

File: src/pages/HomeDashBoard.jsx (hunk 13 above)
Comment (abhi1906singh) [reply src/pages/HomeDashBoard.jsx]: Resolved sir

File: src/context/NoteState.jsx (hunk 12 above)
Comment (abhi1906singh) [reply src/context/NoteState.jsx]: Removed

File: src/comonents/InputField.jsx (hunk 10 above)
Comment (abhi1906singh) [reply src/comonents/InputField.jsx]: Resolved

**Pull request merged**